marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/ConstraintsSpec.hs

```diff
@@ -486,6 +492,109 @@ spec = do
              $ ensureMinUtxo protocolTestnet (inAddress, inValue)
          pure $ selectLovelace outValue `shouldBe` maximum [selectLovelace inValue, selectLovelace expected]
 
+  describe "balanceTx" do
+    prop "tx should balance if coin selection succeeded and no Plutus failure occurred" \(SomeTxConstraints marloweVersion constraints) -> do
+      marloweContext <- genSimpleMarloweContext marloweVersion constraints
+
+      -- We MUST dictate the distribution of wallet context assets, default
+      -- generation only tests with empty wallets!
+      maxLovelace <- choose (0, 40_000_000)
```

Comment:
Do we really want to generate a wallet with a tiny amount of Lovelace (< ~1,000,000)? No real wallet could contain that little. I suppose this is okay because the test should be conditional on the transaction being balanceable.

Reply:
It's generating a small amount of wallets that result in unbalanceable transactions

```  balanceTx
    tx should balance if coin selection succeeded and no Plutus failure occurred [v]     
      +++ OK, passed 100 tests:
      95% balancing succeeded
       5% non-balanceable test cases
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
But we don't have to leave it that way if we don't like it.